packages/oauth/src/install-provider.ts

```diff
@@ -336,48 +436,65 @@ export class InstallProvider {
   ): Promise<void> {
     let code: string;
     let flowError: string;
-    let state: string;
+    let stateInQueryString: string;
     try {
       if (req.url !== undefined) {
         // Note: Protocol/ host of object are not necessarily accurate
         // and shouldn't be relied on
         // intended only for accessing searchParams only
-        const searchParams = InstallProvider.extractSearchParams(req);
+        const searchParams = extractSearchParams(req);
         flowError = searchParams.get('error') as string;
         if (flowError === 'access_denied') {
           throw new AuthorizationError('User cancelled the OAuth installation flow!');
         }
         code = searchParams.get('code') as string;
-        state = searchParams.get('state') as string;
+        stateInQueryString = searchParams.get('state') as string;
         if (!code) {
           throw new MissingCodeError('Redirect url is missing the required code query parameter');
         }
-        if (this.stateVerification && !state) {
+        if (this.stateVerification && !stateInQueryString) {
           throw new MissingStateError('Redirect url is missing the state query parameter. If this is intentional, see options for disabling default state verification.');
         }
       } else {
         throw new UnknownError('Something went wrong');
       }
       // If state verification is enabled, attempt to verify, otherwise ignore
-      if (this.stateVerification && this.stateStore) {
-        // eslint-disable-next-line no-param-reassign
-        installOptions = await this.stateStore.verifyStateParam(new Date(), state);
+      if (this.stateVerification) {
+        try {
+          if (this.legacyStateVerification) {
+            // This mode is not enabled by default
+            // This option is for some of the existing developers that need time for migration
+            this.logger.warn('Enabling legacyStateVerification is not recommended as it does not properly work for OAuth CSRF protection. Please consider migrating from directly using InstallProvider#generateInstallUrl() to InstallProvider#handleInstallPath() for serving the install path.');
```

Comment:
Do we perhaps need to provide a bit more guidance on what this migration looks like for developers? How would they know what migration looks like? Is there a doc we should link them to?

Reply:
@filmaj 
The reason why I added this is to cover the use cases where developers cannot immediately enable the cookie validation for some reason. For instance, when we receive questions in GitHub issues (or directly from customers), we may reactively suggest this option for the developers (as a workaround). Thus, when an app enables this option, the maintainers of the app should know what they are doing and why. The migration timing and remaining tasks should be managed on the developer side.

That being said, we can add this option somewhere in docs. I don't think that we should highlight this one, though.

Lastly, this is not a manual migration. As the current behavior of the `@slack/oauth` is not correct in terms of OAuth flow handling in general. Thus, this is a bug fix, not a breaking change that requires developers' action.

Reply:
OK, sounds good.